front/app/modules/commercial/insights/admin/containers/Insights/Edit/InputsTable/Actions.tsx

```diff
@@ -0,0 +1,254 @@
+  const toggleDropdown = () => {
+    setDropdownOpened(!dropdownOpened);
+  };
+
+  const [categorySelection, setCategorySelection] = useState(new Set<string>());
+  const toggleCategory = (categoryId: string) => () => {
+    if (categorySelection.has(categoryId)) {
+      const newSelection = new Set(categorySelection);
+      newSelection.delete(categoryId);
+      setCategorySelection(newSelection);
+    } else {
+      const newSelection = new Set(categorySelection);
+      newSelection.add(categoryId);
+      setCategorySelection(newSelection);
+    }
+  };
+
+  const [processing, setProcessing] = useState(false);
+  // assigns selectedCategories to selectedInputs
+  const assign = async () => {
+    if (selectedInputs.size > 0 && categorySelection.size > 0) {
+      try {
+        setProcessing(true);
+        await batchAssignCategories(
+          viewId,
+          [...selectedInputs],
+          [...categorySelection]
+        );
+      } catch {
+        // do nothing
+      }
+      setProcessing(false);
+      setDropdownOpened(false);
+    }
+  };
+
+  if (isNilOrError(categories)) {
+    return null;
+  }
+
+  const selectedCategory = categories?.find(
+    (category) => category.id === query.category
+  );
+
+  // unassigns categoryFilter from selectedInputs, with confirmation
+  const unassign = async () => {
+    if (selectedInputs.size > 0 && selectedCategory) {
+      const deleteMessage = formatMessage(messages.deleteFromCategories, {
+        categoryName: selectedCategory.attributes.name,
+        selectedCount: selectedInputs.size,
+      });
+      if (window.confirm(deleteMessage)) {
+        try {
+          setProcessing(true);
+          await batchUnassignCategories(
+            viewId,
+            [...selectedInputs],
+            [selectedCategory.id]
+          );
+        } catch {
+          // do nothing
+        }
+        setProcessing(false);
+        setDropdownOpened(false);
+      }
+    }
+  };
+
+  const otherCategories = categories.filter(
+    (category) => category.id !== selectedCategory?.id
+  );
+
+  return (
+    <ActionButtons className={className} data-testid="insightsTableActions">
+      {selectedInputs.size > 0 && (
+        <>
+          {otherCategories.length > 0 && (
+            <ActionButtonWrapper>
+              <Button onClick={toggleDropdown} buttonStyle="admin-dark-text">
+                <StyledIcon name="moveFolder" />
+                <FormattedMessage {...messages.bulkAssignCategory} />
+              </Button>
+
+              <Dropdown
+                top="45px"
+                opened={dropdownOpened}
+                onClickOutside={toggleDropdown}
+                content={
+                  <DropdownList>
+                    {otherCategories.map((category) => (
+                      <DropdownListItem
+                        key={category.id}
+                        onClick={toggleCategory(category.id)}
```

Comment:
One little problem I noticed here is that because this onClick action is on the container and not the checkbox itself, if you click directly on it - it does not get selected.

Reply:
Fixed, good catch. Working on the checkbox I noticed it should support mixed state, I'll log into tech debt to properly move the old mixed boxes to the compoent library